### PR TITLE
use react router version 4 final

### PR DESCRIPTION
**Pull request by zemirco**

### CODE REVIEW

File: src/js/bottomnavigation/test/test.js [hunk 1]
```diff
@@ -24,34 +17,47 @@ const _window = {
...
+          <Route path='/one' component={() => <h1>One</h1>} />
+          <Route path='/two' component={() => <h1>Two</h1>} />
         </BottomNavigation>
-      </LocationBroadcast>
-      )
+      </MemoryRouter>
+    )
     const node = wrapper.find('.BottomNavigation-content').at(0).node
     node.scrollTop = 40
     wrapper.find('.BottomNavigation-menu-item').at(0).simulate('click', {})
```
Comment (lipp) [inline src/js/bottomnavigation/test/test.js]: replacing the simulated click with:

`wrapper.instance().history.push('/two')`

seems to work

Comment (lipp) [inline src/js/bottomnavigation/test/test.js]: this is stupid and does not test what is intended to be....

**Comment by lipp** [comment]
fixed bottom navigation scroll top in branch (based on this): 

https://github.com/HBM/md-components/tree/fix-bottomnavigation-scroll-top

merge/rebase/whatever into this PR? or do it separetly?

**Comment by lipp** [comment]
still seeing this on console

```
Warning: Hash history cannot PUSH the same path; a new entry will not be added to the history stack
```

**Comment by zemirco** [comment]
yes, they are on it

https://github.com/ReactTraining/history/pull/179

**Comment by zemirco** [comment]
and

> But you should only get this warning in development. If you generate your production build correctly (using NODE_ENV=production) you shouldn't see this warning in production.

https://github.com/ReactTraining/react-router/issues/4467

**Pull request merged**